local/core/discoverDeviceCodec.go

```diff
@@ -16,7 +16,7 @@ func (c DiscoverDeviceCodec) ContentFormat() message.MediaType { return message.
 
 // Encode propagates the payload without any conversions.
 func (c DiscoverDeviceCodec) Encode(v interface{}) ([]byte, error) {
-	return nil, fmt.Errorf("not supported")
+	return nil, MakeUnavailable(fmt.Errorf("not supported"))
```

Comment:
Don't use Unavailable for not transient error:
```
    // Unavailable indicates the service is currently unavailable.
    // This is a most likely a transient condition and may be corrected
    // by retrying with a backoff. Note that it is not always safe to retry
    // non-idempotent operations.
    //
    // See litmus test above for deciding between FailedPrecondition,
    // Aborted, and Unavailable.
    //
    // This error code will be generated by the gRPC framework during
    // abrupt shutdown of a server process or network connection.
    Unavailable Code = 14
```

Reply:
So what do you suggest I should use instead, in this case ?

Reply:
UNIMPLEMENTED -The operation is not implemented or is not supported/enabled in this service.